Source/WTF/wtf/text/StringImpl.h

```diff
@@ -1324,6 +1327,44 @@ inline Ref<StringImpl> StringImpl::removeCharacters(const Predicate& findMatch)
     return removeCharactersImpl(characters16(), findMatch);
 }
 
+inline Ref<StringImpl> StringImpl::createByReplacingInCharacters(const LChar* characters, unsigned length, UChar target, UChar replacement, unsigned indexOfFirstTargetCharacter)
+{
+    ASSERT(indexOfFirstTargetCharacter < length);
+    if (isLatin1(replacement)) {
+        LChar* data;
+        LChar oldChar = target;
+        LChar newChar = replacement;
+        auto newImpl = createUninitializedInternalNonEmpty(length, data);
+        memcpy(data, characters, indexOfFirstTargetCharacter * sizeof(LChar));
```

Comment:
Just realized that the `sizeof(LChar)` here is a bit silly. I would have left that out.